### PR TITLE
Add --noenter4b flag for SPI update

### PR DESCRIPTION
**Pull request by aranikam**
For spi operations for targets that don't support 4B ……mode

### CODE REVIEW

File: examples/htool.c [hunk 1]
```diff
@@ -422,6 +426,8 @@ static const struct htool_cmd CMDS[] = {
                 {HTOOL_FLAG_VALUE, 'n', "length",
                  .desc = "the number of bytes to read"},
                 {HTOOL_POSITIONAL, .name = "dest-file"},
+                {HTOOL_FLAG_BOOL, 'b', "noenter4b", "false",
```
Comment (gyandreev) [inline examples/htool.c:429]: Maybe we can change the character to something else and reserve b for binary?

File: examples/htool_spi_proxy.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ struct htool_spi_proxy {
   bool is_4_byte;
 };
 
-int htool_spi_proxy_init(struct htool_spi_proxy* spi, struct libhoth_device* dev);
+int htool_spi_proxy_init(struct htool_spi_proxy* spi, struct libhoth_device* dev, bool noenter4b);
```
Comment (gyandreev) [inline examples/htool_spi_proxy.h:28]: Variable name in definition does not match declaration. Is that intended for documentation reasons?

Comment (gyandreev) [reply examples/htool_spi_proxy.h:28]: Actually, they are contradictory. `noenter4b` != `is_4_byte`

**Comment by daimeng-wang** [comment]
We should also consider situations where the Hoth is 4B mode only and doesn't support Enter/Exit4B

File: examples/htool.c [hunk 3]
```diff
@@ -433,6 +474,10 @@ static const struct htool_cmd CMDS[] = {
                 {HTOOL_FLAG_VALUE, 's', "start", "0", .desc = "start address"},
                 {HTOOL_FLAG_BOOL, 'v', "verify", "true"},
                 {HTOOL_POSITIONAL, .name = "source-file"},
+                {HTOOL_FLAG_VALUE, 'a', "address_mode", "3B/4B",
```
Comment (gyandreev) [inline examples/htool.c:477]: Nit suggestion, can we move the FLAG_VALUE to before the POSITIONAL? Just for readability so the positional param does not get buried behind other flags.

**Review by cjevans-google — APPROVED** [review]

**Pull request merged**